src/storm/mesos/MesosCommon.java

```diff
@@ -39,6 +40,22 @@
   public static final String SUPERVISOR_ID = "supervisorid";
   public static final String ASSIGNMENT_ID = "assignmentid";
 
+  
+  public static String ipAddressFromAssignmentId(String assignmentId) {
+        int last = assignmentId.lastIndexOf("-");
+        String ip = assignmentId.substring(last + 1);
+        LOG.info("IPAddress: " + ip);
```

Comment:
will change to debug. assignmentid is used to kill the topology so should be a problem. I tested with/without setting the topology.mesos.worker.prefix and it worked.